### DISCUSSION TITLE
Keycloak 16 Offline sessions preloading

### DISCUSSION THREAD

**Discussion by kaushi5054** [keycloak/keycloak#24116] [Archive]
Hi All,

In my project we are using keycloak (**16.1.1**) as IDP. We are facing issue when server restarted all the sessions will be remove. to resolve this Im planning to enable the Offline sessions. I only found the enabling [cmd](https://www.keycloak.org/docs/latest/server_admin/#offline-sessions-preloading) for _22.04_ (latest version).

<img width="1065" alt="image" src="https://github.com/keycloak/keycloak/assets/20614311/92916957-b3fe-4bfb-a525-67dcf73abc7b">

Since I'm searching last few days I couldn't find the cmd to enable it. 

I'm looking for advice, how to enable the Offline sessions for **keycloak 16.1.1.**

Thank you.

**Comment by sschu** [comment]
Offline sessions are always enabled. A user gets an offline sessison when a client requests a token with scope=offline_access.
However, this does not do what you expect. It gives you the ability to use offline tokens that you can use to get new access tokens when the user does not have a session. It does not mean normal sessions are stored persistently. Do do this, you would need a setup where you store the sessions in an external Infinispan and possibly persist them there.

**Reply by kaushi5054** [reply]
Hi sschu,

Thank you so much for your feedback let me try your options.

Thank you.

**Comment by kaushi5054** [comment]
Once we added below env it's work perfectly .

CACHE_OWNERS_COUNT=2
CACHE_OWNERS_AUTH_SESSIONS_COUNT=2
JGROUPS_DISCOVERY_PROTOCOL=JDBC_PING

JGROUPS_DISCOVERY_PROPERTIES="datasource_jndi_name=java:jboss/datasources/KeycloakDS"
